libraries/Microsoft.Bot.Builder.Ai.Translation/LocaleConverterMiddleware.cs

```diff
@@ -1,94 +1,75 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+
+using Microsoft.Bot.Schema;
 using System;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.Bot.Schema;
 
 namespace Microsoft.Bot.Builder.Ai.Translation
 {
     /// <summary>
-    /// Middleware that translates from the input locale to a specified locale.
+    /// Middleware to convert messages between different locales specified
     /// </summary>
     public class LocaleConverterMiddleware : IMiddleware
     {
         private readonly ILocaleConverter _localeConverter;
         private readonly string _toLocale;
-        private readonly Func<ITurnContext, string> _getUserLocale;
-        private readonly Func<ITurnContext, Task<bool>> _setUserLocale;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="LocaleConverterMiddleware"/> class.
+        private readonly IPropertyAccessor<string> _userLocaleProperty;
+
+        /// <summary>
+        /// Constructor for developer defined detection of user messages
         /// </summary>
-        /// <param name="getUserLocale">Delegate for getting the user locale.</param>
-        /// <param name="checkUserLocaleChanged">Delegate that returns true if the locale was
-        /// changed (implements logic to change locale by intercepting the message).</param>
-        /// <param name="toLocale">The target locale.</param>
-        /// <param name="localeConverter">The locale converter to use.</param>
-        /// <exception cref="ArgumentNullException">Thrown when any of the parameters is null.</exception>
-        public LocaleConverterMiddleware(Func<ITurnContext, string> getUserLocale, Func<ITurnContext, Task<bool>> checkUserLocaleChanged, string toLocale, ILocaleConverter localeConverter)
+        /// <param name="userLocaleProperty">PropertyAccessor for the users preferred locale</param>
+        /// <param name="toLocale">Target Locale</param>
+        /// <param name="localeConverter">An ILocaleConverter instance</param>
+        public LocaleConverterMiddleware(IPropertyAccessor<string> userLocaleProperty, string toLocale, ILocaleConverter localeConverter)
         {
             _localeConverter = localeConverter ?? throw new ArgumentNullException(nameof(localeConverter));
-            if (string.IsNullOrEmpty(toLocale))
-            {
-                throw new ArgumentNullException(nameof(toLocale));
-            }
-            else if (!localeConverter.IsLocaleAvailable(toLocale))
-            {
-                throw new ArgumentNullException("The locale " + nameof(toLocale) + " is unavailable");
-            }
-
+            if (string.IsNullOrEmpty(toLocale))
```

Comment:
This branch is missing the stylecop updates. 

Please rebase this branch against master to pickup the many (!) change that have been made.